terraform_fmt.sh

```diff
@@ -46,44 +48,77 @@ parse_cmdline_() {
   done
 }
 
-terraform_fmt_() {
-
-  declare -a paths
-  declare -a tfvars_files
-
-  index=0
-
-  for file_with_path in "${FILES[@]}"; do
+function terraform_fmt_ {
+  local -r args="$1"
+  shift 1
+  local -a -r files=("$@")
+  # consume modified files passed from pre-commit so that
+  # hook runs against only those relevant directories
+  local index=0
+  for file_with_path in "${files[@]}"; do
     file_with_path="${file_with_path// /__REPLACED__SPACE__}"
 
-    paths[index]=$(dirname "$file_with_path")
-
+    dir_paths[index]=$(dirname "$file_with_path")
+    # TODO Unique part
     if [[ "$file_with_path" == *".tfvars" ]]; then
       tfvars_files+=("$file_with_path")
     fi
-
+    #? End for unique part
     ((index += 1))
   done
 
-  for path_uniq in $(echo "${paths[*]}" | tr ' ' '\n' | sort -u); do
-    path_uniq="${path_uniq//__REPLACED__SPACE__/ }"
+  # allow hook to continue if exit_code is greater than 0
+  # preserve errexit status
+  shopt -qo errexit && ERREXIT_IS_SET=true
+  set +e
+  local final_exit_code=0
 
-    (
-      cd "$path_uniq"
-      terraform fmt "${ARGS[@]}"
-    )
+  # run hook for each path
+  for dir_path in $(echo "${dir_paths[*]}" | tr ' ' '\n' | sort -u); do
```

Comment:
This will play bad if any `dir_path` has a space in it
```suggestion
  declare -A dir_paths_tmp
  for item in "${dir_paths[@]}"; do dir_paths_tmp["$item"]+=1; done
  dir_paths=("${!dir_paths_tmp[@]}")
  unset dir_paths_tmp
  for dir_path in "${dir_paths[@]}"; do
```

Reply:
This means we also probably can get rid of `__REPLACED__SPACE__` from wherever it comes from.

Reply:
`dir_path` will have no spaces in their path, during

```bash
    file_with_path="${file_with_path// /__REPLACED__SPACE__}"

    dir_paths[index]=$(dirname "$file_with_path")
```

Also, I didn't understand what the code in Suggested change does. 
I will not replace things that looks pretty straight forward with things that need tons of comments

Reply:
> Also, I didn't understand what the code in Suggested change does.

It replaces external deps like `tr` and `sort` with bash builtin functionality.

>  I will not replace things that looks pretty straight forward with things that need tons of comments

Makes sense. It was literally a _suggestion_. I'm not pushing any suggestion I propose. 
If peer engineer says what you said, this means the suggestion should not be accepted!
Thanks for being mindful and straight-thinking 👍🏻